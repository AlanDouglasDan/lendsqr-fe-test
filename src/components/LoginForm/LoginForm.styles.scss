@use "variables";

.welcomeText {
  font-size: 40px;
  font-weight: 700;
  color: variables.$dark-blue;
}

.welcomeText2 {
  font-size: 20px;
  font-weight: 300;
  color: variables.$dim-grey;
  margin-bottom: 60px;
}

.minWidth {
  width: 447px;
}

.border-right-0 {
  border-right: 0px;
}

.border-left-0 {
  border-left: 0px;
}

.showText {
  color: variables.$green !important;
  font-size: 12px;
  letter-spacing: 0.1em;
  cursor: pointer;
  font-weight: 600;
}

.form-control,
.input-group-text {
  border-color: rgba(84, 95, 125, 0.15);
  padding: 16px;
  font-size: 14px;
  color: variables.$dim-grey;
  border-width: 2px;
}

.input-group {
  border: 2px solid rgba(84, 95, 125, 0.15);
}

.form-control::placeholder {
  color: variables.$dim-grey;
  font-size: 14px;
  opacity: 0.6;
}

.form-control:focus,
.input-group:focus-within {
  border-color: variables.$green !important;
  box-shadow: none;
}

.submitButton {
  background-color: variables.$green;
  height: 48px;
  border: none;
  border-radius: 8px;
  letter-spacing: 0.1em;
}

.forgotPassword {
  color: variables.$green;
  margin-bottom: 30px;
  letter-spacing: 0.1em;
  cursor: pointer;
  font-weight: 500;
}
