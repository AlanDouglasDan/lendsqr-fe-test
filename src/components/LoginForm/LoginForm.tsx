import React, { useState } from "react";
import { useNavigate } from "react-router-dom";
import { toast } from "react-toastify";

import "./LoginForm.styles.scss";

const LoginForm = () => {
  const [showPassword, setShowPassword] = useState<boolean>(false);

  const [email, setEmail] = useState<string>("");
  const [password, setPassword] = useState<string>("");

  const navigate = useNavigate();

  const handleSubmit = () => {
    if (!email || !password) toast.error("Please fill all fields");
    else {
      toast.success("Login Successful");
      navigate("users");
    }
  };

  return (
    <div className="w-100">
      <div className="welcomeText">Welcome!</div>

      <div className="welcomeText2">Enter details to login.</div>

      <div className="minWidth w-lg-100">
        <div className="form-group">
          <input
            type="email"
            className="form-control"
            id="email"
            placeholder="Enter email"
            onChange={(e) => setEmail(e.target.value)}
            value={email}
            data-testid="email"
          />
        </div>

        <div className="form-group">
          <div className="input-group my-4 form-control p-0">
            <input
              type={!showPassword ? "password" : "text"}
              className="form-control border-0"
              placeholder="Password"
              aria-label="password"
              onChange={(e) => setPassword(e.target.value)}
              value={password}
              data-testid="password"
            />
            <span
              className="input-group-text bg-white showText m-0 border-0"
              onClick={() => setShowPassword(!showPassword)}
            >
              {!showPassword ? "SHOW" : "HIDE"}
            </span>
          </div>
        </div>

        <div className="forgotPassword">FORGOT PASSWORD?</div>

        <button
          type="submit"
          className="w-100 submitButton text-white"
          onClick={() => handleSubmit()}
        >
          LOG IN
        </button>
      </div>
    </div>
  );
};

export default LoginForm;
