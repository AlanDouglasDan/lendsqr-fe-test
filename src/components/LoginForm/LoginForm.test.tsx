import { BrowserRouter } from "react-router-dom";
import { render, fireEvent, screen } from "@testing-library/react";
import { toast } from "react-toastify";

import LoginForm from "./LoginForm";

describe("Login Form Component Test", () => {
  it("Logging in with valid credentials", () => {
    render(
      <BrowserRouter>
        <LoginForm />
      </BrowserRouter>
    );

    // Mock the success toast function
    toast.success = jest.fn();

    // Fill in the email and password fields
    fireEvent.change(screen.getByTestId("email"), {
      target: { value: "example@example.com" },
    });
    fireEvent.change(screen.getByTestId("password"), {
      target: { value: "password123" },
    });

    // Trigger the login button click
    fireEvent.click(screen.getByRole("button", { name: "LOG IN" }));

    // Assert that the success toast is displayed
    expect(toast.success).toHaveBeenCalledWith("Login Successful");
  });

  it("Logging in with invalid credentials", () => {
    render(
      <BrowserRouter>
        <LoginForm />
      </BrowserRouter>
    );

    // Mock the error toast function
    toast.error = jest.fn();

    // Fill in the email and password fields
    fireEvent.change(screen.getByTestId("email"), {
      target: { value: "" },
    });
    fireEvent.change(screen.getByTestId("password"), {
      target: { value: "" },
    });

    // Trigger the login button click
    fireEvent.click(screen.getByRole("button", { name: "LOG IN" }));

    // Assert that the error toast is displayed
    expect(toast.error).toHaveBeenCalledWith("Please fill all fields");
  });

  it("Login Form Component Snapshot", () => {
    const { asFragment } = render(
      <BrowserRouter>
        <LoginForm />
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
