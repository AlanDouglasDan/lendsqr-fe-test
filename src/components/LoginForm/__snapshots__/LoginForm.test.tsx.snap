// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Form Component Test Login Form Component Snapshot 1`] = `
<DocumentFragment>
  <div
    class="w-100"
  >
    <div
      class="welcomeText"
    >
      Welcome!
    </div>
    <div
      class="welcomeText2"
    >
      Enter details to login.
    </div>
    <div
      class="minWidth w-lg-100"
    >
      <div
        class="form-group"
      >
        <input
          autocomplete="off"
          class="form-control undefined"
          data-testid="email"
          id="email"
          placeholder="Enter email"
          type="email"
          value=""
        />
      </div>
      <div
        class="form-group"
      >
        <div
          class="input-group my-4 form-control p-0"
        >
          <input
            aria-label="password"
            class="form-control border-0"
            data-testid="password"
            placeholder="Password"
            type="password"
            value=""
          />
          <span
            class="input-group-text bg-white showText m-0 border-0"
          >
            SHOW
          </span>
        </div>
      </div>
      <div
        class="forgotPassword"
      >
        FORGOT PASSWORD?
      </div>
      <button
        class="w-100 submitButton text-white"
        type="submit"
      >
        LOG IN
      </button>
    </div>
  </div>
</DocumentFragment>
`;
